content/upcoming/enry.md

```diff
@@ -0,0 +1,182 @@
+
+The detection process must be split in several steps that form a chain or sequence of strategies. This strategies need the file's name and content and a list of candidates(a list of possible languages for the file). Strategies are typed functions with the next signature:
+```
+type Strategy func(filename string, content []byte, candidates []string) (languages []string)
+```
+
+Strategies try to guess the language of the file based on this arguments and a specific characteristic. For example the default strategies enry uses are:
+```
+var DefaultStrategies = []Strategy{
+	GetLanguagesByModeline,
+	GetLanguagesByFilename,
+	GetLanguagesByShebang,
+	GetLanguagesByExtension,
+	GetLanguagesByContent,
+	GetLanguagesByClassifier,
+}
+```
+You can see how the strategies look for Modeline, Filename, Shebang... characteristics of a file that can be representative of the language it uses. Some strategies needs to parse the content and perhaps apply heuristics while others, however,  do its job only with the filename.
+
+The way the strategies chain works is at follows:
+* A strategy try to get the language from a file.
+* It can result in: no language detected, one language detected, more than one language detected.
+* If there's no language, call to the next strategy.
+* If there are more than one languages, they are appended to a list of candidates and it is given to the next strategy.
+* If there is exactly one language, it is returned as the detected language and the process stops.
+
+When the detection process is falling through the strategies without get a only one language as a outcome, the last strategy `GetLanguagesByClassifier` is obviously reached . This strategy deserves take a look at it.
+
+`GetLanguagesByClassifier` internally uses an object of the type:
+```
+type Classifier interface {
+	Classify(content []byte, candidates map[string]float64) (languages []string)
+}
+```
+
+This interface allow you to implement your own classifier and use it throw the functions:
+```
+func GetLanguageBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (language string, safe bool)
+func GetLanguagesBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (languages []string)
+```
+
+*enry*'s default classifier implementation is a bayesian classifier which follows the same implementation *linguist* uses. It assigns scores to the candidates regarding to its probability, with the highest score assigned to the most likely candidate.
+
+By the way, strategies have a `GetLanguage-` version too that returns only a language, and boolean to indicate the sureness of this result. You can use this functions independently for whatever you want.
+
+If you want to custom *enry*'s default strategies and classifier to use your own implemented strategies and classifiers, you can do it assign them to the variables:
```

Comment:
s/implemented//
s/do it assign them to the variables:/do so by assigning them to the following variables:/